checker/src/abstract_value.rs

```diff
@@ -377,6 +377,40 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         } else if self.is_top() {
             other
         } else {
+            if let Expression::Or { left: x, right: y } = &self.expression {
+                // (x || y) && x == x
+                // (x || y) && y == y
+                if *x == other || *y == other {
+                    return other;
+                }
+                if let Expression::LogicalNot { operand } = &other.expression {
+                    //((x || y) && (!x)) = y
```

Comment:
nit: use == here and elsewhere for consistency. Also space to // and unnecessary parentheses.

Reply:
I am trying to avoid overloading ==. The = here is a meta operator. I'll change it to -> so that it is less confusing.

Reply:
Just asking for consistency. Above you used ==